build_tag_push.py

```diff
@@ -1,101 +1,155 @@
-	print image_id
-	#os.environ['IMAGE_ID'] = image_id  # This -q flag retrieves the last image ID (Doesn't work... TypeError)
-	#os.system("docker run $IMAGE_ID")
-	#os.system("docker run docker images -q $IMAGE_NAME") -> Nope.
-	#os.system("docker images -q $IMAGE_NAME")
-	#os.system("IMAGE_ID=$(docker images -q $IMAGE_NAME)")
-	os.system("echo done")
-	#os.system('docker run $IMAGE_ID')
-
-	##########################################
-	# NOT FINISHED HERE -- WILL PICK UP ON IT.
-	# Left off on trying to store the Image ID 
-	# as an environment variable to use that when 
-	# running the ID. 
-	##########################################
-
-
-# Step 4 - Tags the image with the current commit hash & pushed to Docker Hub = NOT COMPLETE.
-#
-#
-#
-#
-#
-#
-#
-#
-#
-#
-#
+import yaml
+import sys
+import os
+import subprocess
+from shutil import copyfile
+import argparse
+import tempfile
+
+def main():
+    parser = argparse.ArgumentParser(description='Usage: python3'
+                                     'build_tag_push.py <docker_hub_name'
+                                     '<app_yml_path')
+
+    parser.add_argument('docker_hub_name', type=str,
+                        help='DockerHub namespace')
+
+    parser.add_argument('app_yml_path', type=str,
+                        help='Repository path to MVP Studio file')
+
+    parser.add_argument('--ignore_test_errors', type=bool,
+                        help='Boolean value if you want to ignore'
+                        'test errors.')
+
+    args = parser.parse_args()
+
+    data_from_app_yml = parse_app_file(args.app_yml_path)
+
+    create_context_directory(data_from_app_yml['docker_context'],
+                             args.app_yml_path)
+
+    build_check_app(data_from_app_yml, args.app_yml_path,
+                    args.ignore_test_errors)
+
+    build_docker_image(data_from_app_yml, args.app_yml_path)
+
+    tag_docker_image(data_from_app_yml['name'], args.docker_hub_name)
+
+    push_docker_image(data_from_app_yml['name'], args.docker_hub_name)
+
+
+def parse_app_file(filename):
+    '''
+    This function is using the yaml built in to parse the file
+    which can be done by searching for the key in the dictionary
+    for the yaml file.
+    '''
+    with open(filename, "r") as file_contents:
+        dict_of_contents = yaml.load(file_contents)
+
+    return dict_of_contents
+
+
+def build_check_app(data_from_app_yml, args_app_yml_path,
+                    args_ignore_test_errors):
+    '''
+    This is building and testing their app. Ignoring the return code.
+    '''
+    path_to_build_and_test = args_app_yml_path
+    os.chdir(os.path.dirname(path_to_build_and_test))
+    subprocess.check_call(['bash', data_from_app_yml['build']])
+
+    result = subprocess.call(['bash', data_from_app_yml['test']])
+
+    if result != 0:
+        if args_ignore_test_errors:
+            print("ERROR: Tests failes.")
+        else:
+            Assert.ThrowsException('Unit tests failed.')
```

Comment:
I don't know what `Assert` is - I don't see anything imported and I don't see a class called `Assert` that you write. Does this work?

The standard way to throw in Python would just be `raise RuntimeError('Unit tests failed.')`